gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/AllergyIntoleranceMapperTest.java

```diff
@@ -346,6 +352,42 @@ public void testMapAllergyWithDrugTermText() {
         assertThat(allergyIntolerance.getCode().getText()).isEqualTo(DISPLAY_NAME_IN_VALUE);
     }
 
+    @Test
+    public void When_AllergyIntoleranceWithQualifierAndOriginalText_Expect_NotesContainsEpisodicity() {
+        when(codeableConceptMapper.mapToCodeableConcept(any(CD.class)))
+                .thenReturn(defaultCodeableConcept());
+
+        var ehrExtract = unmarshallEhrExtract("drug_allergy_with_qualifier_and_original_text.xml");
+
+        List<AllergyIntolerance> allergyIntolerances = allergyIntoleranceMapper.mapResources(ehrExtract, getPatient(),
+                getEncounterList(), PRACTISE_CODE);
+
+        var allergyIntolerance = allergyIntolerances.get(0);
+
+        assertFixedValues(allergyIntolerance);
+        assertThat(allergyIntolerance.getNote().get(0).getText())
+                .isEqualTo(EPISODICITY_WITH_ORIGINAL_TEXT_NOTE_TEXT);
+    }
+
+    @Test
+    public void When_AllergyIntoleranceWithQualifierAndWithoutOriginalText_Expect_NotesContainsEpisodicity() {
+        when(codeableConceptMapper.mapToCodeableConcept(any(CD.class)))
+                .thenReturn(defaultCodeableConcept());
+
+        var ehrExtract = unmarshallEhrExtract("drug_allergy_with_qualifier_without_original_text.xml");
+
+        List<AllergyIntolerance> allergyIntolerances = allergyIntoleranceMapper.mapResources(ehrExtract, getPatient(),
+                getEncounterList(), PRACTISE_CODE);
+
+        var allergyIntolerance = allergyIntolerances.get(0);
+
+        assertFixedValues(allergyIntolerance);
```

Comment:
I feell like this is a violation of the single assert rule.

Reply:
Yeah it doesn't need to be there, I've just accidentally left in it